resources/sdk/webmessagingjava/scripts/generate-custom-markdown.ts

```diff
@@ -0,0 +1,348 @@
+                if (constructorDeclaration.param) {
+                    fileOutput += `
+
+### Parameters
+
+
+| Name | Type | Description  | Notes |
+| ------------- | ------------- | ------------- | ------------- |
+`
+                    for (const param of constructorDeclaration.param) {
+                        fileOutput += `| **${param.values[0]}** | **${param.values[1]}**| ${param.values[2]} |\n`
+                    }
+                    fileOutput += "{: class=\"table-striped\"}\n\n"
+                }
+            }
+
+            fileOutput += "\n\n<h1>Methods</h1>\n\n"
+
+            // Output method descriptions
+            for (const methodDeclaration of parsedFile) {
+                if (methodDeclaration.type.length > 1 && methodDeclaration.type[1] === "class" || methodDeclaration.type[1] === "interface") {
+                    continue
+                }
+
+                let returnType
+                if (methodDeclaration.type[0] === "private" || methodDeclaration.type[0] === "public" || methodDeclaration.type[0] === "protected") {
+                    returnType = methodDeclaration.type[1]
+                } else {
+                    returnType = methodDeclaration.type[0]
+                }
+
+                fileOutput += `<a name="${methodDeclaration.name}${methodDeclarationCount[methodDeclaration.name].shift()}"></a>
+
+# **${methodDeclaration.name}**
+
+
+
+> ${returnType || ""} ${methodDeclaration.name}(`
+
+                if (methodDeclaration.param) {
+                    fileOutput += methodDeclaration.param[0].values[0]
+                    for (let i = 1; i < methodDeclaration.param.length; i++) {
+                        fileOutput += `, ${methodDeclaration.param[i].values[0]}`
+                    }
+                }
+                fileOutput += `)
+
+${methodDeclaration.description.replace(" \n\n", "")}`
+                if (methodDeclaration.param) {
+                    fileOutput += `
+
+### Parameters
+
+
+| Name | Type | Description  | Notes |
+| ------------- | ------------- | ------------- | ------------- |
+`
+                    for (const param of methodDeclaration.param) {
+                        fileOutput += `| **${param.values[0]}** | **${param.values[1]}**| ${param.values[2]} |\n`
+                    }
+                    fileOutput += "{: class=\"table-striped\"}\n\n"
+                }
+
+                if (returnType) {
+                    fileOutput += `
+### Return type
+
+**${returnType}**
+
+`
+                }
+            }
+
+            if (this.saveMarkdownFile) {
+                const savePath = path.join(this.saveMarkdownFile, this.fileName) + ".md"
+                console.log(`Writing markdown to ${savePath}`)
+                fs.writeFileSync(savePath, fileOutput)
+            }
+
+        }
+
+        catch (err) {
+            process.exitCode = 1;
+            console.log(err);
+        }
+    };
+
+    parseFile(file, prefix, options) {
```

Comment:
man this is a massive method.  We might want to spend some to refactor it into chunks.  Also should we explicitly define these functions as private or public within the class.  At the very least a comment above each method wiould help break this up.